.github/workflows/ci.yml

```diff
@@ -5,43 +5,108 @@ on:
     - cron: 0 2 * * 1-5
   pull_request:
 
+env:
+  COVERAGE: '0'
+  EXT_PCOV_VERSION: '1.0.6'
+
 jobs:
   tests:
     name: Tests (PHP ${{ matrix.php }})
     runs-on: ubuntu-latest
     container:
-      image: php:${{ matrix.php }}
+      image: php:${{ matrix.php }}-alpine
+      options: >-
+        --tmpfs /tmp:exec
     strategy:
       # https://github.community/t5/GitHub-Actions/GitHub-Actions-Matrix-options-dont-work-as-documented/m-p/29558
       matrix:
         php:
           - '7.2'
           - '7.3'
+        include:
+          - php: '7.3'
+            coverage: true
         dependency:
           - ''
           - '--prefer-lowest && composer remove --dev --no-progress --no-update --ansi league/flysystem-bundle'
       fail-fast: false
     steps:
       - name: Checkout
         uses: actions/checkout@v1
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache
+          key: ${{ matrix.php }}-composer-${{ matrix.dependency }}-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            ${{ matrix.php }}-composer-${{ matrix.dependency }}-
+
       - name: Install system packages
         run: |
-          apt-get update
-          apt-get install -y --no-install-recommends \
-            unzip
+          apk add \
+            unzip \
+
+      - name: Install pcov PHP extension
+        if: matrix.coverage && matrix.dependency == ''
+        run: |
+          apk add $PHPIZE_DEPS
+          pecl install pcov-$EXT_PCOV_VERSION
+          docker-php-ext-enable pcov
+
       - name: Disable PHP memory limit
         run: echo 'memory_limit=-1' >> /usr/local/etc/php/php.ini
+
       - name: Install Composer
         run: curl -sS -o - https://raw.githubusercontent.com/composer/getcomposer.org/ba13e3fc70f1c66250d1ea7ea4911d593aa1dba5/web/installer | php -- --install-dir=/usr/local/bin --filename=composer --quiet
+
       - name: Update project dependencies
         run: composer update --no-progress --no-suggest --ansi ${{ matrix.dependency }}
-      - name: Run PHPUnit tests
-        run: vendor/bin/phpunit --colors=always
+
       - name: Run phpstan tests
         if: matrix.dependency == ''
         run: vendor/bin/phpstan analyze
+
       - name: Run phpstan tests lowest
         if: matrix.dependency != ''
         run: vendor/bin/phpstan analyze -c phpstan.neon.lowest.dist
+
       - name: Run php-cs-fixer tests
         run: vendor/bin/php-cs-fixer fix --dry-run --diff
+
+      - name: Enable code coverage
+        if: matrix.coverage && matrix.dependency == ''
+        run: echo '::set-env name=COVERAGE::1'
+
+      - name: Run PHPUnit tests
+        run: |
+          mkdir -p build/logs/phpunit
```

Comment:
Isn't it possible to split this test in 1 tests, using `if` syntax?:
```
- name: Run PHPUnit tests with coverage
  if: matrix.coverage
  run: vendor/bin/phpunit --coverage-clover ...

- name: Run PHPUnit tests
  if: !matrix.coverage
  run: vendor/bin/phpunit ...
```

Reply:
I'll try something.